src/Mvc/Mvc.Core/src/ModelBinding/Metadata/SystemTextJsonValidationMetadataProvider.cs

```diff
@@ -57,7 +57,9 @@ public void CreateValidationMetadata(ValidationMetadataProviderContext context)
 
         if (string.IsNullOrEmpty(propertyName))
         {
-            propertyName = _jsonNamingPolicy.ConvertName(context.Key.Name!);
+            propertyName = context.Key.Name is { } contextKeyName
```

Comment:
```suggestion
            propertyName = context.Key.Name is string contextKeyName
```

Nit: To make it clear what the parameter type expectation for `ConvertName` is.